### PR TITLE
Add employees note api

### PR DESCRIPTION
**Pull request by JMetherell**
Add entity model, controller, service and transformer

Resolves:
SFA-1078

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/Kind.java [hunk 1]
```diff
@@ -14,7 +14,8 @@
     INTANGIBLE_ASSETS_NOTE("small-full-accounts#intangible-assets-note"),
     SMALL_FULL_STATEMENT("small-full-accounts#statements"),
     STOCKS_NOTE("small-full-accounts#stocks-note"),
-    TANGIBLE_ASSETS_NOTE("small-full-accounts-note#tangible-assets");
+    TANGIBLE_ASSETS_NOTE("small-full-accounts-note#tangible-assets"),
+    EMPLOYEES_NOTE("small-full-accounts-note#employees-note");
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/Kind.java]: Do we need 'note' in both parts of the link? We should be consistent with the format of these. For reference, according to the spec the tangible link should be of the format: `small-full-accounts-note#tangible-assets` so I'd expect this to be `small-full-accounts-note#employees`

Comment (JMetherell) [reply src/main/java/uk/gov/companieshouse/api/accounts/Kind.java]: It really should be consistent - looks like stocks and intangible don't have note in the prefix either, but the spec says they should, so I'll update them all to match the spec - "small-full-accounts-note#foo"

File: src/main/java/uk/gov/companieshouse/api/accounts/transformer/EmployeesTransformer.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        if (rest.getCurrentPeriod() != null) {
+            BeanUtils.copyProperties(rest, employeesDataEntity);
+            CurrentPeriodEntity currentPeriodEntity = new CurrentPeriodEntity();
+            BeanUtils.copyProperties(rest.getCurrentPeriod(), currentPeriodEntity);
+            employeesDataEntity.setCurrentPeriodEntity(currentPeriodEntity);
+        }
+
+        if (rest.getPreviousPeriod() != null) {
+            BeanUtils.copyProperties(rest, employeesDataEntity);
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/transformer/EmployeesTransformer.java]: Do we need this call in both 'if' blocks?

File: src/main/java/uk/gov/companieshouse/api/accounts/transformer/EmployeesTransformer.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        return employeesEntity;
+    }
+
+    @Override
+    public Employees transform(EmployeesEntity entity) {
+
+        Employees employees = new Employees();
+        EmployeesDataEntity employeesDataEntity;
+
+        if (entity.getData() != null) {
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/transformer/EmployeesTransformer.java]: Will this ever be `null`?

Comment (JMetherell) [reply src/main/java/uk/gov/companieshouse/api/accounts/transformer/EmployeesTransformer.java]: It wouldn't be in the normal course of events, but if the data in mongo isn't there as expected, the null check handles it gracefully.

File: src/test/java/uk/gov/companieshouse/api/accounts/transformer/EmployeesTransformerTest.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        CurrentPeriod currentPeriod = new CurrentPeriod();
+        currentPeriod.setAverageNumberOfEmployees(AVERAGE_NUMBER_OF_EMPLOYEES);
+        currentPeriod.setDetails(DETAILS);
+        return currentPeriod;
+    }
+
+
+    private PreviousPeriod createPreviousPeriodRestObject() {
+        PreviousPeriod previousPeriod = new PreviousPeriod();
+        previousPeriod.setAverageNumberOfEmployees(AVERAGE_NUMBER_OF_EMPLOYEES);
```
[the remark this answers is not in the document]
Comment (JMetherell) [reply src/test/java/uk/gov/companieshouse/api/accounts/transformer/EmployeesTransformerTest.java]: Updated as suggested

**Pull request merged**